src/ticket.c

```diff
@@ -182,15 +183,27 @@ void save_committed_tkt(struct ticket_config *tk)
 static void ext_prog_failed(struct ticket_config *tk,
 		int start_election)
 {
-	/* Give it to somebody else.
-	 * Just send a VOTE_FOR message, so the
-	 * others can start elections. */
-	if (leader_and_valid(tk)) {
-		save_committed_tkt(tk);
-		reset_ticket(tk);
-		ticket_write(tk);
-		if (start_election) {
-			ticket_broadcast(tk, OP_VOTE_FOR, OP_REQ_VOTE, RLT_SUCCESS, OR_LOCAL_FAIL);
+	if (tk->mode != TICKET_MODE_MANUAL) {
+		/* Give it to somebody else.
+	 	 * Just send a VOTE_FOR message, so the
+		 * others can start elections. */
+		if (leader_and_valid(tk)) {
+			save_committed_tkt(tk);
+			reset_ticket(tk);
+			ticket_write(tk);	
+			if (start_election) {
+				ticket_broadcast(tk, OP_VOTE_FOR, OP_REQ_VOTE, RLT_SUCCESS, OR_LOCAL_FAIL);
+			}
+		}
+	} else {
+		/* There is not much we can do now because
+		 * the manual ticket cannot be relocated.
+		 * Just warn the user. */
+		if (tk->leader == local) {
+			save_committed_tkt(tk);
+			reset_ticket(tk);
+			ticket_write(tk);	
+			log_error("external test failed on the specified machine, cannot acquire a manual ticket");
```

Comment:
We should probably make external tests an invalid configuration for manual tickets.

Reply:
do you mean, to disable external tests for manual tickets completely? hm, I am not sure, perhaps this feature may be useful. We could assure that a ticket cannot be granted somewhere, event if someone make a mistake and try to grant it there..

Reply:
Yes. Well, I'm not sure either. The question is one of balance between simplicity of use and trust, i.e. if the user _wants_ to grant the ticket then let them do that no matter what. Perhaps you can discuss that further also with Yan.

Reply:
I'd prefer to honor before-acquire-handler also for manual tickets if it's configured, which means the user would still like to double-check with that. Of course, users should still be able to bypass that by supplying "-F" option, or probably "-FF" if there's already another condition against granting that needs to be bypassed ;-)

Reply:
Wouldn't we want to keep this behavior consistent with automatic tickets?
__Automatic tickets__
* when we try to grant a ticket to a particular site and this fails, Booth will start a new election process and try to grant a ticket somewhere else. It may possibly fail altogether if no site can pass external tests.
* we cannot force the ticket to be granted to the site which we prefer. there is no flag that allows us to skip external tests.
__Manual tickets__
* wouldn't it be better to not be able to grant a ticket to the preferred site as well?
* we don't have automatic election, so after failure the user can do the same as Booth for automatic tickets - try to grant it somewhere else. But also without certainty that there is any site that would accept the ticket.

I agree we could have a flag for forcing granting a ticket (skipping external tests) but perhaps we could apply it on a higher level, for both manual and automatic tickets?

Reply:
Probably yes. Since there could be manual interference for automatic tickets...